### PR TITLE
fix: responsive element not work in ios 15 and below

### PR DESCRIPTION
**Pull request by bualoy-napat**
## Description
The native ResizeObserver is not work with responsive element 

Fixes # (issue) 
STG-418

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: packages/core/src/elements/ResponsiveElement.ts [hunk 1]
```diff
@@ -30,7 +30,15 @@ export type ResizeEvent = CustomEvent<{
  * @returns {void}
  */
 const triggerResize = (entry: ResizeObserverEntry): void => {
-  const { inlineSize: width, blockSize: height } = entry.borderBoxSize[0];
+  let width; let height;
```
Comment (wattachai-lseg) [inline packages/core/src/elements/ResponsiveElement.ts]: This should be better:
```suggestion
  let width: number;
  let height: number;
```

File: packages/core/src/elements/ResponsiveElement.ts [hunk 2]
```diff
@@ -30,7 +30,15 @@ export type ResizeEvent = CustomEvent<{
...
+    width = entry.borderBoxSize[0].inlineSize;
+    height = entry.borderBoxSize[0].blockSize;
+  }
+  else {
+    width = entry.contentRect.width;
+    height = entry.contentRect.height;
+  }
   const event = new CustomEvent('resize', {
     bubbles: false,
     cancelable: false,
```
Comment (wattachai-lseg) [inline packages/core/src/elements/ResponsiveElement.ts:46]: we can shorten `detail` on line 45 as the following: 
```
detail: {
      width,
      height
}
```

**Review by wattachai-lseg — CHANGES_REQUESTED** [review]
just 2 things.

File: packages/core/src/elements/ResponsiveElement.ts (hunk 1 above)
Comment (bualoy-napat) [reply packages/core/src/elements/ResponsiveElement.ts]: Fixed!

File: packages/core/src/elements/ResponsiveElement.ts (hunk 2 above)
Comment (bualoy-napat) [reply packages/core/src/elements/ResponsiveElement.ts:46]: Fixed!

File: packages/core/src/elements/ResponsiveElement.ts [hunk 3]
```diff
@@ -30,13 +30,22 @@ export type ResizeEvent = CustomEvent<{
  * @returns {void}
  */
 const triggerResize = (entry: ResizeObserverEntry): void => {
-  const { inlineSize: width, blockSize: height } = entry.borderBoxSize[0];
+  let width: number;
+  let height: number;
+  if (entry.borderBoxSize?.length > 0) {
```
Comment (wattachai-lseg) [inline packages/core/src/elements/ResponsiveElement.ts:36]: could you add above this line? This should improve code coverage report from SonarCloud.
`/* istanbul ignore else */`

Comment (bualoy-napat) [reply packages/core/src/elements/ResponsiveElement.ts:36]: Done

**Review by wattachai-lseg — COMMENTED** [review]
LGTM. Just in case, let's wait for feedback from Office team first.

**Review by wattachai-lseg — APPROVED** [review]

**Pull request merged**